src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1969,6 +1969,17 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         list_api->reverse(plist, asr_el_type, *module);
     }
 
+    void generate_Type(ASR::expr_t* m_arg) {
+        this->visit_expr_wrapper(m_arg, true);
+        llvm::Value* item = tmp;
+
+        // Get the LLVM type of the value
+        llvm::Type* llvmType = item->getType();
+
+        // Create an LLVM expression for the type
+        tmp = builder->CreateBitCast(tmp, llvmType);
```

Comment:
Also I am not too sure , if this is how generate_Type should have been introduced !? Maybe @czgdp1807 could help me out here .